3rd party: Build a minimum mix

otaviobp opened the pull request:
In this PR I enable mixer to build a minimum mix including only an empty bundle.

Fixes  #632

rchiossi commented on builder/bundles.go:
```diff
@@ -638,7 +638,7 @@ func (b *Builder) buildBundles(set bundleSet, downloadRetries int) error {
 
 	if _, ok := set[b.Config.Swupd.Bundle]; !ok {
-		return fmt.Errorf("couldn't find bundle %q specified in configuration as the update bundle",
+		fmt.Printf("Warning: couldn't find bundle %q specified in configuration as the update bundle. Bundle won't be configured.\n",
```
Rather than having this warning, we still check if `b.Config.Swupd.Bundle` is `""`. This would indicate that there should be no update bundle. Then, whenever we use mixer for 3rd-party, we change config so that Swupd.BUNDLE is empty.

otaviobp replied:
That was my first implementation, but I thought that this would be more strait forward. If you remove the os-core-update from the mix, we don't update that. If you add it, we update.
So it's about the bundles that are selected from the user, not the name of the bundle that should include swupd.
But I'm ok to change that too, if you think it's going to be less error prone

rchiossi commented on builder/bundles.go:
```diff
@@ -624,7 +624,7 @@ func (b *Builder) buildBundles(set bundleSet, downloadRetries int) error {
 
 	if _, ok := set["os-core"]; !ok {
-		return fmt.Errorf("os-core bundle not found")
+		fmt.Printf("Warning: os-core bundle not found. System may not be complete.\n")
```
I don't like having a warning for something that is expected behavior. I'd rather remove it or have it print some sort of "build mode" message, say "Building package mix"/"Building system mix". Those are not proper terms we use in mixer, so we might as well just remove the warning.

otaviobp replied:
ok. I'll remove.
Is there a problem on now enforcing os-core? Should I add a flag or config to make sure no one will create a regular mix without os-core?

rchiossi commented on swupd/create_manifests.go:
```diff
@@ -205,7 +205,7 @@ func processBundles(ui UpdateInfo, c config, numWorkers int) ([]*Manifest, error
 		changedFiles, added, deleted := bundle.linkPeersAndChange(oldM, ui.minVersion)
 		// if nothing changed, skip
-		if changedFiles == 0 && added == 0 && deleted == 0 && !changedIncludes {
+		if changedFiles == 0 && added == 0 && deleted == 0 && !changedIncludes && bundle.Name != "full" {
```
Do you know why mixer crashes without full?

otaviobp replied:
It crashes if there's no manifests in a list of manifests in create_manifests. But anyway, we need to create the full manifest for this release even if it hasn't changed, right?

otaviobp commented:
> I'm not sure about the --skip-update-index option. It looks like a quick hack, It could actually be a viable option to get rid of that bundle in upstream.

First, yes, it's a hack. But my implementation isn't removing the update-index. if you already have that it will just move it forward. I can change it to make it empty and in this case it can be used to get rid of the bundle upstream. But if this is the case, can't we just enforce it to be empty? And then we don't even need a flag?

gtkramer commented:
If I'm understanding this PR correctly, would this `--skip-update-index` option be passed on `mixer build bundles` and `mixer build update`?  And would it work without supplying 3rd party content?

otaviobp commented:
V2 - Making some requested changes:
 - Remove error messages on building a mix without os-core or os-core-update
 - Removed the commit that added a os-core-update-index flag because this is going to be handled in #636.
 - Fixed the test to work with os-core-update-index generation
